apps/ex_nvr/lib/ex_nvr/remote_storages/snapshot_uploader.ex

```diff
@@ -0,0 +1,128 @@
+defmodule ExNVR.RemoteStorages.SnapshotUploader do
+  use GenServer, restart: :transient
+
+  require Logger
+
+  alias ExNVR.RemoteStorage
+  alias ExNVR.{Devices, RemoteStorages}
+  alias ExNVR.RemoteStorages.Store
+
+  def start_link(options) do
+    GenServer.start_link(__MODULE__, options)
+  end
+
+  @impl true
+  def init(options) do
+    Logger.metadata(device_id: options[:device].id)
+    Logger.info("Start snapshot uploader")
+    send(self(), :init_config)
+    {:ok, %{device: options[:device], schedule: %{}, opts: []}}
+  end
+
+  @impl true
+  def handle_info(:init_config, %{device: device} = state) do
+    snapshot_config = device.snapshot_config
```

Comment:
A better approach here is to not access the fields directly but to add a function in the device
```elixir
def snapshot_config(device) do
# you can now return the fields you want with a parsed schedule too
# this way even if the schema changes, we can still expose the same fields through this function
end
```